resource/src/routes/route.mustache

```diff
@@ -1,26 +1,33 @@
+{{#isUser}}
+const { Router } = require('express');
+{{/isUser}}
 const { createRouter } = require('../utils/router');
 const {{camelCasePlural}} = require('../controllers/{{snakeCasePlural}}');
 
+{{#isUser}}
 const serialize = {{camelCase}} => {
   const data = Object.assign({}, {{camelCase}});
-  {{#isUser}}
   delete data.password;
-  {{/isUser}}
   return data;
 };
 
-{{#isUser}}
-const template = {
-  post: {
-    '/signin': function *(req, res) {
-      const { user, token } = yield {{camelCasePlural}}.signin(req.body.email, req.body.password);
-      res.json({ token, user: serialize(user) });
-    },
-    '/signout': function *(req, res) {
-      res.json(yield {{camelCasePlural}}.signout(req.body));
-    }
-  }
-};
+const customRouter = new Router()
+  .post('/signin', function *(req, res) {
+    const { user, token } = yield {{camelCasePlural}}.signin(req.body.email, req.body.password);
+    res.json({ token, user: serialize(user) });
+  })
+  .post('/signout', function *(req, res) {
+    res.json(yield {{camelCasePlural}}.signout(req.body));
+  });
+
+const defaultRouter = createRouter({{camelCasePlural}}, serialize);
+customRouter.socket = defaultRouter.socket;
+
+customRouter.use(defaultRouter);
+
+module.exports = customRouter;
```

Comment:
Hmm, i think you didn't really need to do this.  at the moment the `createRouter` returns a normal express router, and `/signin`, `/signout` don't interfere with any REST api methods. so you could just do something like this:

```js
module.exports = createRouter({{camelCasePlural}}, serialize)
  .post('/signin', function *(req, res) {
    const { user, token } = yield {{camelCasePlural}}.signin(req.body.email, req.body.password);
    res.json({ token, user: serialize(user) });
  })
  .post('/signout', function *(req, res) {
    res.json(yield {{camelCasePlural}}.signout(req.body));
  });
```